aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -28,7 +28,8 @@ static inline std::tuple<Tensor, Tensor> _lu_det_P_diag_U(const Tensor& self) {
   std::tie(lu, pivs, infos) = at::_lu_with_info(self, /*pivot=*/true, /*check_errors=*/false);
   TORCH_CHECK(infos.ge(0).all().item<uint8_t>(), "Invalid argument passed to lu");
   auto n = self.size(-1);
-  auto num_exchanges = (at::arange(1, n + 1, pivs.options()) != pivs).sum(-1, /*keepdim=*/false, /*dtype=*/self.scalar_type()).fmod_(2);
+  auto num_exchanges = (at::arange(1, n + 1, pivs.options().dtype(at::kLong)) != pivs.to(at::kLong))
```

Comment:
are you converting to kLong to avoid possible overflows? Even in that case setting dtype=at::kLong in `sum` should be enough

Reply:
`n` is also `long` I suppose.

Reply:
Maybe actually a better fix would be to return `pivs` as integers, as they are always integers.

Reply:
Hmm, now `pivs` are actually integers, from master:
```python
In [1]: import torch

In [2]: x = torch.rand(3, 3, dtype=torch.cdouble)

In [3]: x
Out[3]: 
tensor([[0.3119+0.0148j, 0.6229+0.6629j, 0.0268+0.5217j],
        [0.7130+0.8608j, 0.8461+0.4621j, 0.3453+0.4870j],
        [0.5092+0.6666j, 0.7671+0.6573j, 0.1088+0.0076j]],
       dtype=torch.complex128)

In [4]: x.lu()
Out[4]: 
(tensor([[ 0.7130+0.8608j,  0.8461+0.4621j,  0.3453+0.4870j],
         [ 0.1882-0.2065j,  0.3683+0.7506j, -0.1388+0.5013j],
         [ 0.7500+0.0296j,  0.3839-0.0066j, -0.0858-0.5612j]],
        dtype=torch.complex128),
 tensor([2, 2, 3], dtype=torch.int32))

```

Reply:
Sorry, just changing the `dtype` in `sum` is indeed sufficient :)